code-samples/oauth/simple-oauth-webserver/index.js

```diff
@@ -1,58 +1,43 @@
-const http = require('http');
-const fs = require('fs');
-const url = require('url');
+const express = require('express');
+const app = express();
 const fetch = require('node-fetch');
 
-const port = 53134;
-
-http.createServer((req, res) => {
-	let responseCode = 404;
-	let content = '404 Error';
-
-	const urlObj = url.parse(req.url, true);
-
-	if (urlObj.query.code) {
-		const accessCode = urlObj.query.code;
-		const data = {
-			client_id: 'your client id',
-			client_secret: 'your client secret',
-			grant_type: 'authorization_code',
-			redirect_uri: 'your redirect uri',
-			code: accessCode,
-			scope: 'the scopes',
-		};
-
-		fetch('https://discord.com/api/oauth2/token', {
-			method: 'POST',
+app.get('/', async (request, response) => {
+	let access_token;
+	let token_type;
+	if (request.query.code) {
+	    const accessCode = request.query.code;
+	    const data = {
+	        client_id: 'client_id',
+	        client_secret: 'client_secret',
+	        grant_type: 'authorization_code',
+	        redirect_uri: 'http://localhost:3000/',
+	        code: accessCode,
+	        scope: 'identify',
+	    };
+
+	    const res = await fetch('https://discord.com/api/oauth2/token', {
+	        method: 'POST',
 			body: new URLSearchParams(data),
 			headers: {
 				'Content-Type': 'application/x-www-form-urlencoded',
 			},
-		})
-			.then(discordRes => discordRes.json())
-			.then(info => {
-				console.log(info);
-				return info;
-			})
-			.then(info => fetch('https://discord.com/api/users/@me', {
-				headers: {
-					authorization: `${info.token_type} ${info.access_token}`,
-				},
-			}))
-			.then(userRes => userRes.json())
-			.then(console.log);
-	}
-
-	if (urlObj.pathname === '/') {
-		responseCode = 200;
-		content = fs.readFileSync('./index.html');
+		});
+		const info = await res.json();
```

Comment:
```suggestion
		const { token_type: tokenType, access_token: accessToken } = await res.json();
```
As well as update the 2 lines below this and the 2 `let` declarations at the top of the function.